retry.go

```diff
@@ -161,15 +161,23 @@ type clientInterface interface {
 }
 
 type retryHTTP struct {
-	ctx      context.Context
-	client   clientInterface
-	req      requestFunc
-	method   string
-	fullURL  *url.URL
-	headers  map[string]string
-	body     []byte
-	timeout  time.Duration
-	raise4XX bool
+	ctx         context.Context
+	client      clientInterface
+	req         requestFunc
+	method      string
+	fullURL     *url.URL
+	headers     map[string]string
+	body        []byte
+	bodyCreator func() ([]byte, error)
+	timeout     time.Duration
+	raise4XX    bool
+}
+
+func (r *retryHTTP) createBody() ([]byte, error) {
```

Comment:
I think this shouldn't be responsibility of this struct. I would rather have it as a part of request (maybe instead of `requestFunc` there should be another struct that has this `requestFunc` and body). Although this might be a part of a bigger refactoring.

Reply:
Considering the next comment - this is obsolete and removed.